go/vt/vttablet/tabletserver/tx_engine.go

```diff
@@ -75,7 +75,11 @@ type TxEngine struct {
 	// transition while creating new transactions
 	beginRequests sync.WaitGroup
 
-	twopcEnabled        bool
+	// twopcEnabled is the flag value of whether the user has enabled twopc or not.
+	twopcEnabled bool
+	// twopcAllowed is wether it is safe to allow two pc transactions or not.
+	// If the primary tablet doesn't run with semi-sync we set this to false, and disallow any prepared calls.
+	twopcAllowed        bool
```

Comment:
I think we can manage this with one field only and would not need 2 fields

Reply:
No we can't. One is a flag and the other is whether semi-sync is enabled or not.

Reply:
We can make it to an enum of something if you want to merge them.

Reply:
flag is separate from the tx engine struct. So, for the tx engine it does not matter
config should be true and semi-sync should be on, only then distributed transactions are allowed on that tablet.
I believe it could be possible with one field.

Reply:
One field will never be enough. Let's say the user turned on semi-sync on a primary that wasn't running semi-sync before. Now do you turn on your single field? We don't know if it was false only because semi-sync was off, or if the flag was also off or not.

Reply:
We can always refer to the flag config, we are not going to change the flag config field